File: CODE-I/V-Skills/Node/NodeConsoleApp1/services/DivisionService.js
export default class DivisionService {
    getHighest(a, b) {
        if (a === 0 || b === 0) {
            throw new Error("Cannot divide by zero");
        }

        const resultA = a / b;
        const resultB = b / a;

        const max = Math.max(resultA, resultB);
        const winner = resultA > resultB ? "a divided by b" : (resultB > resultA ? "b divided by a" : "tie");

        return { max, winner };
    }
}

File: CODE-I/V-Skills/ASP.NETWebApplication1/Domain/Services/DivisionService.cs
using ASP.NETWebApplication1.Domain.Interfaces;
using ASP.NETWebApplication1.Domain.Models;

namespace ASP.NETWebApplication1.Domain.Services
{
    public class DivisionService : IDivisionService
    {
        public DivisionResult GetHighest(double a, double b)
        {
            if (a == 0 || b == 0)
                throw new ArgumentException("Cannot divide by zero");

            double result1 = a / b;
            double result2 = b / a;

            return new DivisionResult
            {
                Result1 = result1,
                Result2 = result2,
                Max = Math.Max(result1, result2),
                Winner = result1 > result2 ? "a divided by b" : result2 > result1 ? "b divided by a" : "tie"
            };

        }
    }
}
